### PR TITLE
Various spelling fixes

### PR DESCRIPTION
**Pull request by jsoref**
Generated by https://github.com/jsoref/spelling `f`; to maintain your repo, please consider `fchurn`

These changes do not impact program logic

split from #628 per https://github.com/Gnucash/gnucash/pull/628#issuecomment-573331361

### CODE REVIEW

**Comment by jralls** [comment]
@cstim Did you examine every line to be sure?

File: common/debug/valgrind/valgrind-libgda.supp [hunk 1]
```diff
@@ -37,7 +37,7 @@
 #     if Free: name of free-ing fn)
 
 {
-	libgda permanant memory for gda_paramlist_dtd
+	libgda permanent memory for gda_paramlist_dtd
```
Comment (gjanssens) [inline common/debug/valgrind/valgrind-libgda.supp]: I don't know enough of how valgrind works to evaluate whether the name of the suppression matters at all. So I don't know if it's safe to fix the name (though I would assume it is?).

File: libgnucash/backend/sql/escape.cpp [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 /*
  * FILE:
- * esacpe.c
+ * escape.c
```
Comment (gjanssens) [inline libgnucash/backend/sql/escape.cpp]: The filename has become `escape.cpp`, though that's obviously not a spelling fix...

File: libgnucash/backend/xml/io-gncxml-v1.cpp [hunk 3]
```diff
@@ -657,7 +657,7 @@ glist_kvp_value_end_handler (gpointer data_for_children,
         sixtp_child_result* cr = (sixtp_child_result*) lp->data;
         KvpValue* kvp = (KvpValue*) cr->data;
 
-        /* children are in reverse chron order, so this fixes it. */
+        /* children are in reverse cron order, so this fixes it. */
```
Comment (gjanssens) [inline libgnucash/backend/xml/io-gncxml-v1.cpp]: No, chron here is shorthand for chronological, not referring to the cron tool.

File: libgnucash/backend/xml/sixtp-utils.cpp [hunk 4]
```diff
@@ -113,7 +113,7 @@ concatenate_child_result_chars (GSList* data_from_children)
 
     g_return_val_if_fail (name, NULL);
 
-    /* child data lists are in reverse chron order */
+    /* child data lists are in reverse cron order */
```
Comment (gjanssens) [inline libgnucash/backend/xml/sixtp-utils.cpp]: Same here.

**Comment by gjanssens** [comment]
@jralls I did scroll through all changes to verify they are not changing function or variable names. They are all in comments and non-parsed text so that should be ok. I have not verified every single fix for (spelling) correctness. In my cursory scan most seemed obvious fixes anyway, but I don't consider myself authoritative enough in English to evaluate on that level.

File: libgnucash/backend/xml/io-gncxml-v1.cpp (hunk 3 above)
Comment (jsoref) [reply libgnucash/backend/xml/io-gncxml-v1.cpp]: would you object to `chronological`?

File: common/debug/valgrind/valgrind-libgda.supp (hunk 1 above)
Comment (jsoref) [reply common/debug/valgrind/valgrind-libgda.supp]: The first line is the name/description https://wiki.wxwidgets.org/Valgrind_Suppression_File_Howto#Suppression_file_syntax

It is effectively a comment.

Comment (gjanssens) [reply common/debug/valgrind/valgrind-libgda.supp]: Ok, that's what I thought. I just wanted to be sure this string is not referenced anywhere else, like in command line parameters or in other suppression files (not necessarily part of the gnucash repo).

File: libgnucash/backend/xml/io-gncxml-v1.cpp (hunk 3 above)
Comment (gjanssens) [reply libgnucash/backend/xml/io-gncxml-v1.cpp]: No, that's fine.

**Comment by jralls** [comment]
@gjanssens The only spelling issues I found so far that I disagree with are forcing `colour` to `color` and `implementor` to `implementer`. In the first case I see no reason to change British spellings to US ones and  in the second my dictionaries all indicate that both spellings are acceptable. I've only managed to look at about 1/8 of the changes and I don't think that chasing spelling is a good use of my time, your time, or frankly anyone else's time.

File: libgnucash/backend/xml/io-gncxml-v1.cpp (hunk 3 above)
Comment (jsoref) [reply libgnucash/backend/xml/io-gncxml-v1.cpp]: https://github.com/Gnucash/gnucash/pull/634/files#diff-d2acbd5839cd84ca0ea14906500fa669R660

File: libgnucash/backend/xml/sixtp-utils.cpp (hunk 4 above)
Comment (jsoref) [reply libgnucash/backend/xml/sixtp-utils.cpp]: https://github.com/Gnucash/gnucash/pull/634/files#diff-d529c408a55014101ea7b7d3c05ff140R116

File: libgnucash/backend/sql/escape.cpp (hunk 2 above)
Comment (jsoref) [reply libgnucash/backend/sql/escape.cpp]: Do you want this as a PR? https://github.com/Gnucash/gnucash/compare/master...jsoref:spelling-escape?expand=1

(I'm not going to make additional PRs w/o someone confirming...)

File: CMakeLists.txt [hunk 5]
```diff
@@ -591,7 +591,7 @@ if (XCODE_VERSION)
   set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE ${LIBDIR_BUILD}/gnucash)
 endif()
 
-# For binarines
+# For binarnies
```
Comment (dennisschagt) [inline CMakeLists.txt]: `binarnies` -> `binaries`?

Comment (jsoref) [reply CMakeLists.txt]: https://github.com/Gnucash/gnucash/pull/634/files#diff-af3b638bc2a3e6c650974192a53c7291R594

**Comment by jralls** [comment]
We've decided that blanket fixing typos, spelling errors, and forcing American spellings is a bit obnoxious and likely to discourage contributions. It's too bad that you did the work without discussing it with us first.

**Pull request closed without merging**